### PR TITLE
Junos Regex and config load fixes. Adds another example. 

### PR DESCRIPTION
**Pull request by nitinsoniism**
-Fixes a possible bug where LoadConfig always overrides current config.
Now user can merge with the existing config instead of overriding.

-Fixes junos shell and root-shell regex to adapt to the wider range of
junos devices.

Examples os shell prompts:
start shell:
* %
* [vrf:foo] regress@EVOvFOOBAR_RE0-re0:~$

start shell user root:
* root@vMX1_RE:/var/home/regress #
* [vrf:foo] root@EVOvFOOBAR_RE0-re0:~#

Tested these regexes here: https://regex101.com/

-Also added a LoadConfig example as it didn't exist in any of the
examples.

### CODE REVIEW

**Comment by carlmontanari** [comment]
Hey @nitinsoniism thanks for opening this, and good catch on the `replace` flag!

I edited things a little bit:

1. just added some spacing to appease linter
2. modified the patterns a tick to include start line anchor, and to add the `PatternNotContains` to the "normal" shell priv level.

I also have a PR to add this into scrapli (python version) [here](https://github.com/carlmontanari/scrapli/pull/170)

If this looks all good to you I will get it merged!

Thanks again for this work!

Carl

**Comment by nitinsoniism** [comment]
Thanks, Carl. 
Everything looks good except PatternNotContains part of a normal "start shell". It may not work if a root user starts a low privilege shell using "start shell". We might still see pattern "root". Let's remove this as of now.

Thanks for adding start anchors and taking care of the python version.

Scrapligo is good stuff! :-)

We can go ahead with the merge once root shell PatternNotContains is taken care of.

File: examples/network_driver/load_config/main.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	"github.com/scrapli/scrapligo/cfg"
+)
+
+//Pushes config from a file to a file on device and
+//loads it to the device. Cleans up afterwords.
+func main() {
+	d, err := core.NewCoreDriver(
+		"hostname",
+		"juniper_junos",
+		base.WithPort(22),
```
Comment (hellt) [inline examples/network_driver/load_config/main.go:18]: is this needed?

Comment (carlmontanari) [reply examples/network_driver/load_config/main.go:18]: not needed, but cant hurt to have in example to show folks I suppose!

**Comment by carlmontanari** [comment]
> Everything looks good except PatternNotContains part of a normal "start shell". It may not work if a root user starts a low privilege shell using "start shell". We might still see pattern "root". Let's remove this as of now.

The issue with this is that if a user tries to acquire root shell but scrapligo ends up thinking that they only made it to shell priv level we will end up returning an error from `AcquirePriv` method. 

In python this is probably ok (or can make a change to be ok) because we iterate over the priv levels in the privilege dict and that will be guaranteed to be ordered so we could just put the `shell` level *after* `root_shell`. In go (I think?) we cant assume the priv level map will be iterated over in any particular order; which can cause us to erroneously think we are in `shell` rather than `root_shell` if we hit the `shell` priv first and the pattern matches, ultimately causing us to return an error. Hopefully that makes sense, let me know if it doesn't or I'm missing something!

Carl

**Comment by nitinsoniism** [comment]
> > Everything looks good except PatternNotContains part of a normal "start shell". It may not work if a root user starts a low privilege shell using "start shell". We might still see pattern "root". Let's remove this as of now.
> 
> The issue with this is that if a user tries to acquire root shell but scrapligo ends up thinking that they only made it to shell priv level we will end up returning an error from `AcquirePriv` method.
> 
> In python this is probably ok (or can make a change to be ok) because we iterate over the priv levels in the privilege dict and that will be guaranteed to be ordered so we could just put the `shell` level _after_ `root_shell`. In go (I think?) we cant assume the priv level map will be iterated over in any particular order; which can cause us to erroneously think we are in `shell` rather than `root_shell` if we hit the `shell` priv first and the pattern matches, ultimately causing us to return an error. Hopefully that makes sense, let me know if it doesn't or I'm missing something!
> 
> Carl

This makes sense, Carl. I see what you are saying. You can go ahead and merge if all checks pass. It doesn't make sense for a root user to launch a low priv shell hence we should be good.

File: examples/network_driver/load_config/main.go (hunk 1 above)
Comment (nitinsoniism) [reply examples/network_driver/load_config/main.go:18]: @hellt did you mean Port? No, not needed.
Did you mean is NewCoreDriver() call needed? I think we do need a driver for NewCfgDriver() factory.

**Comment by carlmontanari** [comment]
Thanks for the work on this @nitinsoniism !!

Carl

**Pull request merged**